### ISSUE TITLE
persistent builds / continuous integration

### ISSUE THREAD

**Issue by hossbeast** [hossbeast/fab#1]
given target(s), uses inotify to watch for changes to all relevant nodes.

changes are batched into a changeset with a configurable timeout (5s ?)

given a changeset, performs an incremental build to update the target(s)

some kind of HUD view for the output of the latest build

**Comment by hossbeast** [comment]
To go along with this, there needs to be a way of specifying which files should be watched with inotify for changes, and which files should be checked for changes on demand.

For example, inotify may be the best approach for local files, on-demand checking the best option for remote-mounted files (with the caveat that builds in response to changes to those files would have to be be triggered manually)

**Comment by hossbeast** [comment]
Regarding specifying which method to use for watching, a configuration option has been implemented which lets the user specify the invalidation mechanism for portions of the filesystem.

```
filesystem {
  "/" {
    invalidate notify
  }

  "/mnt/remote" {
    invalidate stat
  }
}
```

**Comment by hossbeast** [comment]
"invalidate notify" has also been implemented.

"HUD view for the output of the latest build" won't be implemented as part of v0.6. (fabc, or fab commander)

**Comment by hossbeast** [comment]
done in 4c8dca397fede0c9a2e1734b580dceb94f9f813d
